Updates from porting to crash course.

jw676 opened the pull request:
Updated the codebase to work with IQM and IBM without needing 2 different environments. I tested the code on Odo for both IQM and IBM backends using garnet:mock and fake_sherbrooke, respectively. Feel free to double-check as needed.

The requirements.txt file points to my fork of the revolving quantum_linear_solvers package (which required minor revisions to work), but we should update this once we find a good place for an OLCF maintained version.

jw676 committed:
Changed the main package requirements, so that's it's less strict and can accomadate iqm

jw676 committed:
Modified run sections to make the script IQM and IBM compatible

jw676 committed:
Updated to be compatible with ibm and iqm

jw676 committed:
Updated script to work with IBM and IQM. Cleaned up get_ancillaquibt function for readability, and handle bad simulator results.

jw676 committed:
Convient function for handling plotting in WC challenges.

jw676 committed:
Added sourcable proxy file.

jw676 committed:
Updated documentation to reflect changes in current implementation of the repo.

jw676 committed:
Update requirements for the repo. Added temporary location for the quantum_linear_solvers package, but eventually should point to OLCF maintained repo.

jw676 committed:
Cleaned up file. Updated formating.

jw676 committed:
Fixed minor bugs.

jw676 committed:
Fixed an issue where some lines of code were duplicated. Perhaps from a bad merge?

muralikrishnangm commented on qlsa/func_qc.py:
```diff
@@ -184,6 +197,9 @@ def qc_circ(n_qubits_matrix, classical_solution, args, input_vars):
     if args.backend_type=='ideal':
         print(f'exact_vector/norm:\n{exact_vector/np.linalg.norm(exact_vector)}')
 
+    if args.backend_type=='ideal':
```
Remove this as it is repeated above?

muralikrishnangm commented on qlsa/requirements.txt:
```diff
@@ -1,13 +1,18 @@
 
-qiskit>=1.0
+qiskit
+qiskit
```
Repetitive

muralikrishnangm commented on qlsa/requirements.txt:
```diff
@@ -1,13 +1,18 @@
-qiskit-qasm3-import   
+qiskit-qasm3-import 
+qiskit-iqm
+qiskit-qasm3-import 
```
Repetitive - and below

muralikrishnangm commented on qlsa/requirements.txt:
Shall we add version numbers to be more robust in case future versions are not compatible with current code?

jw676 replied:
That's a good idea. Version conflicts were definitely a headache when I first started working with the repo. I pinned the qiskit versions, but we can provide ranges if that makes more sense in the future.

jw676 committed:
Fixed repetitive line issues and added version numbers.

jw676 commented:
Good catch on all the repetitive lines, Murali. I think some weird bug when resolving merge conflicts might have caused that. The latest commit addresses all of the concerns.

muralikrishnangm reviewed (approved):
Thank you very much for the updates, Jordan!
